api/v1beta2/foundationdbcluster_types.go

```diff
@@ -361,18 +362,29 @@ func (processGroupStatus *ProcessGroupStatus) MarkForRemoval() {
 }
 
 // NeedsReplacement checks if the ProcessGroupStatus has conditions so that it should be removed
-func (processGroupStatus *ProcessGroupStatus) NeedsReplacement(failureTime int) (bool, int64) {
-	var missingTime *int64
-	for _, condition := range conditionsThatNeedReplacement {
-		conditionTime := processGroupStatus.GetConditionTime(condition)
-		if conditionTime != nil && (missingTime == nil || *missingTime > *conditionTime) {
-			missingTime = conditionTime
+func (processGroupStatus *ProcessGroupStatus) NeedsReplacement(failureTime int, taintReplacementTime int) (bool, int64) {
+	var earliestFailureTime *int64
```

Comment:
```suggestion
func (processGroupStatus *ProcessGroupStatus) NeedsReplacement(failureTime int, taintReplacementTime int) (bool, int64) {
	if processGroupStatus.IsMarkedForRemoval() {
		return false, 0
	}

	var earliestFailureTime *int64
```

That way we can remove this check in the if statements below and we don't have to iterate over all conditions.